src/vec_simd.rs

```diff
@@ -67,6 +81,100 @@ pub fn scaled_add(mut u: ArrayViewMut1<f32>, v: ArrayView1<f32>, a: f32) {
     )
 }
 
+#[cfg(all(target_arch = "aarch64", feature = "neon"))]
+pub mod neon {
+    use std::arch::aarch64::*;
+
+    use ndarray::{ArrayView1, ArrayViewMut1};
+
+    use crate::vec_simd::{dot_unvectorized, scale_unvectorized, scaled_add_unvectorized};
+
+    #[inline(always)]
+    #[allow(dead_code)]
+    pub(crate) unsafe fn neon_add(values: float32x4_t) -> f32 {
+        // vaddvq_f32 is missing in std::arch::aarch64?
+        let sums = vadd_f32(vget_high_f32(values), vget_low_f32(values));
+        vget_lane_f32::<0>(sums) + vget_lane_f32::<1>(sums)
+    }
+
+    #[target_feature(enable = "neon")]
+    pub unsafe fn dot(u: ArrayView1<f32>, v: ArrayView1<f32>) -> f32 {
```

Comment:
This function calculates the dot product up until any remaining partition not divisible by 4, those are rolled out after calls to this function?

Reply:
As the last expression in the function

```rust
neon_add(sums) + dot_unvectorized(u, v)
```

Reply:
Huh, guess I was a bit too tired, I totally overlooked that!